src/expm.jl

```diff
@@ -25,25 +26,50 @@ end
 
     v = (a-d)^2 + 4*b*c
 
+    m = (a + d) / 2
+
     if v > 0
-        z = sqrt(v)
-        z1 = cosh(z / 2)
-        z2 = sinh(z / 2) / z
+        # In this case the formulas in the entries of the matrix
+        # are a function of cosh and sinh, and could (in theory)
+        # follow the same code pattern of the other branches (v ≤ 0).
+        # However, cosh and sinh explode with large arguments and
+        # we use the following identity to avoid numerical issues:
+        #
+        # exp(m) * [c₁ * cohs(δ) + c₂ * sinh(δ)] =
+        #   c₁ * (e₊ + e₋) / 2 + c₂ * (e₊ - e₋) / 2
```

Comment:
If you move the `/2`'s as considered below, I guess this comment could be updated to move them as well.